tests/test_main.py

```diff
@@ -3151,6 +3152,33 @@ class Three(One):
     assert getattr(Three, 'foo', None) == ' edited! edited!'
 
 
+def test_shadow_attribute_warn_for_redefined_fields() -> None:
+    """https://github.com/pydantic/pydantic/issues/9107"""
+
+    # A simple class which defines a field
+    class Parent:
+        foo: bool = False
+
+    # When inheriting from the parent class, as long as the field is not defined at all, there should be no warning
+    # about shadowed fields.
+    with warnings.catch_warnings():
+        warnings.simplefilter('error')
```

Comment:
Why is this necessary?

Reply:
Sorry, I misunderstood the `simplefilter()` docs - I have adjusted it now.

Essentially follows [this example](https://docs.python.org/3/library/warnings.html#testing-warnings) from the docs, to ensure that all warnings are captured, then run the code, then check the recorded warnings list is empty.

Reply:
Ahh I see, looks good.